### ISSUE TITLE
Cornice does not support POST operations with colander schemas. 

### ISSUE THREAD

**Issue by shadowmint** [Cornices/cornice#175]
This issue can be summarized by:

Using cornice.Service and the @blah.post decorator with a schema doesn't work properly. 

Specifically: def extract_request_data(request) in utils.py returns 'request.GET, request.headers, body, request.matchdict' in all cases.

_Including_ POST requests. This means that the data passed (in request.POST, see: http://docs.pylonsproject.org/projects/pyramid/en/latest/narr/webob.html) is never loaded or checked against the schema (which is done manually in schemas.validate_colander_schema).

Effectively this means that if you have code like:

```
@hello.post(schema=FooBarSchema)
def post_hello(self):
  return {'status': 'success', 'errors': []}
```

It's basically useless, as the response will always generate errors, regardless of the posted values, as the request.body value is not set in the POST operation.

This is particularly obvious if you try to, for example, create a json application backend and access it via angular or jquery.

Long story short, .post() and schema=BlahSchema are incompatible, and will never result in successful schema validation.

schemas.py appears to be the only places that uses the extract_request_data method in utils (odd, why is isn't it just part of schemas.py?), and it seems moderately trivial to simply add an additional response from extract_request_data for the POST data and another call to _validate_fields() in schemas.py.

...but honestly, reading the documentation I actually have no idea if this is even supposed to be supported or not.

So, if not, feel free to close this ticket (but at least there'll be some documentation of the fact it's not supposed to work...). If so, let me know and I'll submit a push request with the fix.
